client/apps/edit2/components/display/components/preview/social_preview.jsx

```diff
@@ -25,18 +34,18 @@ export const SocialPreview = (props) => {
 
         <div className='edit-display__prev-social--content'>
           <div className='edit-display__prev-social--placeholder'>
-          {article.getThumbnailImage('social_image')
-            ? <img src={crop(article.getThumbnailImage('social_image'), {width: 300, height: 155})} />
+          {image
+            ? <img src={crop(image, {width: 300, height: 155})} />
             : <div className='edit-display__prev--x' />
           }
           </div>
 
           <div className='edit-display__prev-social--meta'>
             <div className='edit-display__prev-social--headline'>
-              {article.getThumbnailTitle('social_title')}
+              {social_title || thumbnail_title}
```

Comment:
Interesting this sort of makes things less reliable now that business logic isn't tied to a model. I was hoping that we could make some utilities or a "model-like" file for managing this logic into real methods. I've seen this kind of problem before where denormalizing business logic creates inconsistencies and makes things difficult to find later on.

Reply:
Right this is why I started the other file.  In the case above, I think that actually using a conditional makes it clearer what is happening, because it isn't actually complex logic. One thing i noticed is that we have model logic that isn't used, or used in one place, but there are cases like the byline where it actually does make sense to move out.

Reply:
I also tried to move some of the model functions into redux actions so that we'd have access to the data and logic everywhere, but it didn't seem to make sense in the end, because it wants you to store the return value from the methods in state.

Reply:
Yeah if the logic in `getThumbnailTitle` isn't used anywhere else I think it's fine to keep here.